corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js

```diff
@@ -458,9 +467,27 @@ hqDefine("cloudcare/js/formplayer/menus/views", function () {
         },
 
         continueAction: function () {
+            var self = this;
+            // clear session storage at the selectedValuesKey when leaving the screen
+            var selectedValuesForScreen = JSON.parse(sessionStorage.selectedValues);
+            selectedValuesForScreen[self.selectedValuesKey] = undefined;
+            sessionStorage.selectedValues = JSON.stringify(selectedValuesForScreen);
+
             FormplayerFrontend.trigger("menu:select", this.selectedCaseIds);
         },
 
+        reconcileSelectedValues: function () {
+            var self = this;
+            if (sessionStorage.selectedValues !== undefined) {
+                var selectedValues = JSON.parse(sessionStorage.selectedValues)
+            } else {
+                var selectedValues = {};
+            }
+            selectedValues[self.selectedValuesKey] = self.selectedCaseIds.join(',');
+            selectedValues['current'] = self.selectedCaseIds.join(',');
```

Comment:
why do we need 'current'? can't we always use `queryKey`?

Reply:
I chose to use `current` to be cleared out after each iteration (which is what we need to do for everything except pagination) and then it checks `queryKey` for when entering the multiSelect case list to get the current selected values. I found I needed both otherwise it was the same issue in pervious PR where it needed to be cleared for every instance except pagination, so just looking by queryKey would break pagination since it would need to be cleared

Reply:
@snopoke okay I had more time to look into this because I was questioning this approach. So I believe querying by `queryKey` would work [here ](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/api.js#L154) to retrieve the items, and then potentially they would not need to be cleared [here](https://github.com/dimagi/commcare-hq/pull/31997/files#diff-71bcd78058a097956160c4925eed5ba9a848392f7e68c69a3d27b4a5ac27e645R70) (would need to test to confirm but that makes since to me. Only issue is that `queryKey` in not accessible within api.js so I would need to add that. I chose to go with `current` instead and then clearing current while maintaining access to the data within `queryKey` if needed for pagination. I see how this is less than ideal, but if you have another suggestion given those limitations or are strongly opposed to this approach let me know

Reply:
@snopoke so I actually did just as described above after I saw your comment about `queryKey` being accessible in `sessionStorage`-- fixed here [14a7e47](https://github.com/dimagi/commcare-hq/pull/31997/commits/14a7e4768884826684ef61206b78f825af028cab) thanks for those insights!

Reply:
Thanks for responding, I was going to try and dig deeper into it today but the changes look great.